src/libraries/Microsoft.PowerFx.Core/Logging/StructuralPrint.cs

```diff
@@ -81,14 +81,21 @@ public override LazyList<string> Visit(FirstNameNode node, Precedence parentPrec
                 return LazyList<string>.Of(sanitizedName);
             }
 
-            var info = _binding?.GetInfo(node);
+            var info = _binding?.GetInfo(node);
             if (info != null && info.Kind != BindKind.Unknown)
             {
                 return LazyList<string>.Of($"#${Enum.GetName(typeof(BindKind), info.Kind)}$#");
-            }
-
+            }
+
             if (node.Ident.AtToken == null)
-            {
+            {
+                var type = _binding?.GetType(node);
+
+                if (type != null && type.AggregateHasExpandedType())
+                {
+                    return LazyList<string>.Of($"#$firstnameexp$#");

```

Comment:
Please check [https://github.com/microsoft/Power-Fx-Dataverse/pull/283](https://github.com/microsoft/Power-Fx-Dataverse/pull/283). This is the PR containing the test cases.

Reply:
Code in this repository should be tested in this repository. We shouldn't rely on other repositories to test that, otherwise we may make changes which would break the other repositories without knowing about it.